cli/cmd/lib_cluster_config.go

```diff
@@ -18,194 +18,156 @@ package cmd
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
-	return nil
+	confirmClusterConfig(clusterConfig, awsCreds)
+
+	return clusterConfig, awsCreds, nil
 }
 
-// awsCreds is what was read from the cluster config YAML, after being passed through setInstallAWSCredentials() (so those credentials should be set)
-func setOperatorAWSCredentials(awsCreds *AWSCredentials) error {
-	// First check env vars
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") != "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") != "" {
-		awsCreds.CortexAWSAccessKeyID = os.Getenv("CORTEX_AWS_ACCESS_KEY_ID")
-		awsCreds.CortexAWSSecretAccessKey = os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY")
-		return nil
-	}
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") == "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") != "" {
-		return errors.New("Only $CORTEX_AWS_SECRET_ACCESS_KEY is set; please run `export CORTEX_AWS_ACCESS_KEY_ID=***`")
-	}
-	if os.Getenv("CORTEX_AWS_ACCESS_KEY_ID") != "" && os.Getenv("CORTEX_AWS_SECRET_ACCESS_KEY") == "" {
-		return errors.New("Only $CORTEX_AWS_ACCESS_KEY_ID is set; please run `export CORTEX_AWS_SECRET_ACCESS_KEY=***`")
+func getUpdateClusterConfig(awsCreds *AWSCredentials) (*clusterconfig.ClusterConfig, error) {
+	cachedClusterConfig := &clusterconfig.ClusterConfig{}
+	err := readClusterConfigFile(cachedClusterConfig, cachedClusterConfigPath)
+	if err != nil {
+		return nil, err
 	}
 
-	// Next check what was read from cluster config YAML
-	if awsCreds.CortexAWSAccessKeyID != "" && awsCreds.CortexAWSSecretAccessKey != "" {
-		return nil
-	}
-	if awsCreds.CortexAWSAccessKeyID == "" && awsCreds.CortexAWSSecretAccessKey != "" {
-		return errors.New(fmt.Sprintf("Only cortex_aws_secret_access_key is set in %s; please set cortex_aws_access_key_id as well", flagClusterConfig))
-	}
-	if awsCreds.CortexAWSAccessKeyID != "" && awsCreds.CortexAWSSecretAccessKey == "" {
-		return errors.New(fmt.Sprintf("Only cortex_aws_access_key_id is set in %s; please set cortex_aws_secret_access_key as well", flagClusterConfig))
+	userClusterConfig := &clusterconfig.ClusterConfig{}
+	err = clusterconfig.SetFileDefaults(userClusterConfig)
+	if err != nil {
+		return nil, err
 	}
 
-	// Default to primary AWS credentials
-	awsCreds.CortexAWSAccessKeyID = awsCreds.AWSAccessKeyID
-	awsCreds.CortexAWSSecretAccessKey = awsCreds.AWSSecretAccessKey
-	return nil
-}
+	if flagClusterConfig == "" {
+		userClusterConfig = cachedClusterConfig
+		err := cr.ReadPrompt(userClusterConfig, clusterconfig.UpdatePromptValidation(false, cachedClusterConfig))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		err = readClusterConfigFile(userClusterConfig, flagClusterConfig)
+		if err != nil {
+			return nil, err
+		}
 
-// awsCreds is what was read from the cluster config YAML
-func setAWSCredentials(awsCreds *AWSCredentials) error {
-	err := setInstallAWSCredentials(awsCreds)
-	if err != nil {
-		return err
-	}
+		userClusterConfig.ClusterName = cachedClusterConfig.ClusterName
```

Comment:
Are these two lines necessary? Will it already have failed if these were changed?